server/datastore/mysql/migrations/tables/20231024105026_AddWindowsMDMTables.go

```diff
@@ -0,0 +1,111 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20231024105026, Down_20231024105026)
+}
+
+func Up_20231024105026(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+-- This table contains the commands, which may target multiple devices.
+CREATE TABLE windows_mdm_commands (
+	-- managed and generated by Fleet, and used as CmdID in the MSMDM messages.
+	command_uuid     VARCHAR(127) NOT NULL,
+	-- the raw XML of the command
+	raw_command      MEDIUMTEXT NOT NULL,
+	-- whether the command was generated from the system or from a user
+	system_origin    TINYINT(1) NOT NULL DEFAULT FALSE,
+	-- the target OMADM URI for the command
+	target_loc_uri   VARCHAR(255) NOT NULL,
+
+	created_at       TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+	updated_at       TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+	PRIMARY KEY (command_uuid)
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create windows_mdm_commands table: %w", err)
+	}
+
+	_, err = tx.Exec(`
+CREATE TABLE windows_mdm_command_queue (
+	host_uuid    VARCHAR(255) NOT NULL,
```

Comment:
I think this should be `device_id` instead of `host_uuid`. MS-MDM only knows about the `device_id` identifier and

Reply:
Yeah the thing is with user-initiated commands, we don't know the `device_id`, only the `host_uuid`. Either way we need some mapping of `device_id` to `host_uuid`, I think it would make sense to store it in `mdm_windows_enrollments` once we get the `host_uuid` for the corresponding `device_id`.

My plan was to have the `host_uuid` in the `windows_mdm_...` tables and when we get only the `device_id`, look it up in that enrollments table, but it works the other way too, we can have `device_id` in the `windows_mdm_...` tables and when we only have the `host_uuid` (i.e. when enqueuing a user command), look it up in the enrollments table.

I'll go with that and update the fields to be called `device_id`, as it's closer to the MSMDM protocol, which is what those tables support.

Reply:
> I think it would make sense to store it in mdm_windows_enrollments once we get the host_uuid for the corresponding device_id.

Yes! I was thinking about this too.

> I'll go with that and update the fields to be called device_id, as it's closer to the MSMDM protocol, which is what those tables support.

good!

Reply:
> I think it would make sense to store it in mdm_windows_enrollments once we get the host_uuid for the corresponding device_id.

+10000 to this